splinepy/microstructure/microstructure.py

```diff
@@ -289,7 +289,9 @@ def create(self, closing_face=None, knot_span_wise=None, **kwargs):
 
         # Prepare the deformation function
         # Transform into a non-uniform splinetype and make sure to work on copy
-        if hasattr(self._deformation_function, "bspline"):
+        if "BSpline" in self._deformation_function.whatami:
+            deformation_function_copy = self._deformation_function.copy()
+        elif hasattr(self._deformation_function, "bspline"):
```

Comment:
What is the difference here exactly.

Reply:
In the case a bspline is passed as a deformation function, there is no conversion. This serves only for readability.